File: examples/c++/stepmotor.cxx
/*
 * Authors: Yevgeniy Kiveisha <yevgeniy.kiveisha@intel.com>
 *          Mihai Tudor Panu <mihai.tudor.panu@intel.com>
 * Copyright (c) 2014 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>
#include <signal.h>
#include <stdio.h>

#include "stepmotor.hpp"
#include "upm_utilities.h"

using namespace std;

int doWork = 1;

void
sig_handler(int signo)
{
    printf("got signal\n");
    if (signo == SIGINT) {
        printf("exiting application\n");
        doWork = 0;
    }
}

int
main(int argc, char** argv)
{
    //! [Interesting]
    upm::StepMotor sensor(2, 3);

    while (doWork) {
        cout << "1 Revolution forward and back at 60 rpm" << endl;
        sensor.setSpeed(60);
        sensor.stepForward(200);
        upm_delay_us(1000000);
        sensor.stepBackward(200);
        upm_delay_us(1000000);

        cout << "1 Revolution forward and back at 150 rpm" << endl;
        sensor.setSpeed(150);
        sensor.stepForward(200);
        upm_delay_us(1000000);
        sensor.stepBackward(200);
        upm_delay_us(1000000);

        cout << "1 Revolution forward and back at 300 rpm" << endl;
        sensor.setSpeed(300);
        sensor.stepForward(200);
        upm_delay_us(1000000);
        sensor.stepBackward(200);
        upm_delay_us(1000000);
    }

    //! [Interesting]
    return 0;
}

File: examples/javascript/stepmotor.js
/*
* Author:  Mihai Tudor Panu <mihai.tudor.panu@intel.com>
* Copyright (c) 2015 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

var lib = require('jsupm_stepmotor');

// Instantiate a stepper motor driver on pins 2 (direction) and 3 (step)
var stepper = new lib.StepMotor(2, 3);

function delay(milliseconds) {
    var startTime = Date.now();
    while (Date.now() - startTime < milliseconds);
}

stepper.quit = function()
{
    stepper = null;
    lib.cleanUp();
    lib = null;
    console.log("Exiting");
    process.exit(0);
};

// The driver is synchronous and blocking
console.log("Rotating 1 revolution forward and back at 60 rpm.");
stepper.setSpeed(60);
stepper.stepForward(200);
delay(1000);
stepper.stepBackward(200);
delay(1000);

console.log("Rotating 1 revolution forward and back at 150 rpm.");
stepper.setSpeed(150);
stepper.stepForward(200);
delay(1000);
stepper.stepBackward(200);
delay(1000);

console.log("Rotating 1 revolution forward and back at 300 rpm.");
stepper.setSpeed(300);
stepper.stepForward(200);
delay(1000);
stepper.stepBackward(200);
delay(1000);

stepper.quit();
